source/Calamari/Commands/FindPackageCommand.cs

```diff
@@ -66,6 +75,10 @@ public override int Execute(string[] commandLineArguments)
                 return 0;
             }
 
+            //Exact package found, so we need to register use and lock it.  We don't lock on partial finds, because there may be too many packages, blocking retention later,
+            //  and we can lock them on apply delta anyway.
+            if (variables.IsPackageRetentionEnabled()) packageJournal.RegisterPackageUse(variables);
```

Comment:
This looks a bit different to the other commands because it add directly to the journal vs via the decorator

Reply:
That's because we only want to add to the journal if we find a full package - otherwise we could lock on too many packages.   That said, I wonder if we should put a lower-value 'soft lock' of some kind on partial finds, to minimise the chance of a package being cleaned up when it may be used as part of a delta transfer - and remove the soft lock on the unused packages once the worker knows which package is actually going to be used for the delta.

Thoughts?